src/runtime/device_interface.cpp

```diff
@@ -594,7 +590,10 @@ WEAK int halide_buffer_copy_already_locked(void *user_context, struct halide_buf
         }
     }
 
-    return err;
+    if (result != halide_error_code_success) {
```

Comment:
We could do that, but I have an idea for a more general fix for calling halide_error() that I'd like to discuss elsewhere

Reply:
> Haven't the device implementations potentially already called halide_error at this point?

But in response to your point:
- It's not currently enforced (or reliably coded) that device implementations do this, hence the possibly-redundant call here. But if they did:
    - If halide_error() calls abort(), having a redundant call doesn't matter because we are dead
    - if halide_error() just logs, we get a redundant log, no biggie